client/packages/common/src/ui/components/buttons/standard/ShrinkableBaseButton.tsx

```diff
@@ -24,28 +25,57 @@ export const StyledShrinkableBaseButton = styled(StyledBaseButton, {
         marginLeft: 8,
       }
     : {},
+
+  '&.MuiButton-root:not(:hover) .MuiSvgIcon-root': {
+    color: color === 'primary' ? translateColor(theme, color) : undefined,
+  },
 }));
 
 interface ShrinkableBaseButtonProps extends ButtonProps {
-  shrink: boolean;
+  label: string;
+  shouldShrink: boolean;
+  shrinkThreshold: 'sm' | 'md' | 'lg' | 'xl';
 }
 
 export const ShrinkableBaseButton = React.forwardRef<
   HTMLButtonElement,
   ShrinkableBaseButtonProps
->(({ shrink = false, onClick, ...props }, ref) => {
-  const { isRtl } = useIntlUtils();
-  return (
-    <StyledShrinkableBaseButton
-      ref={ref}
-      shrink={shrink}
-      size="small"
-      isRtl={isRtl}
-      onClick={onClick}
-      onKeyDown={(event: React.KeyboardEvent<HTMLButtonElement>) => {
-        if (event.code === 'Enter' && !!onClick) onClick({} as any);
-      }}
-      {...props}
-    />
-  );
-});
+>(
+  (
+    { label, onClick, shrinkThreshold, shouldShrink, startIcon, ...props },
+    ref
+  ) => {
+    const { isRtl } = useIntlUtils();
+    const isShrinkThreshold = useIsScreen(shrinkThreshold);
+
+    // On small screens, if the button shouldShrink, then
+    // only display a centred icon, with no text.
+    const shrink = isShrinkThreshold && shouldShrink;
+    // const startIcon = shrink ? null : Icon;
+    const centredIcon = shrink ? startIcon : null;
+    const text = shrink ? null : label;
+
+    return (
+      <Tooltip disableHoverListener={!shrink} title={label}>
```

Comment:
moved some things here, so that I didn't have to duplicate in the BaseButton & LoadingButton